LdapPatch/upgradeLdap.pl

```diff
@@ -0,0 +1,114 @@
+#!/usr/bin/perl
+
+use strict;
+
+use Cwd;
+use Time::localtime qw(ctime);
+
+my $newinstall  = 0;
+our $platform = qx(/opt/zimbra/libexec/get_plat_tag.sh);
+chomp $platform;
+
+my $logFileName = "zmsetup.".getDateStamp().".log";
+my $logfile = "/tmp/".$logFileName;
+open LOGFILE, ">$logfile" or die "Can't open $logfile: $!\n";
+unlink("/tmp/zmsetup.log") if (-e "/tmp/zmsetup.log");
+symlink($logfile, "/tmp/zmsetup.log");
+
+my $ol = select (LOGFILE);
+select ($ol);
+$| = 1;
+
+progress("Operations logged to $logfile\n");
+
+our $ZMPROV = "/opt/zimbra/bin/zmprov -r -m -l";
+our $SU = "su - zimbra -c ";
+
+my $pwuid = getpwuid( $< );
+if ($pwuid eq "zimbra") {
+  $SU = "";
+}
+
+our %options = ();
+
+if(isInstalled("zimbra-ldap")) {
+  installLdapSchema();
+}
+
+sub installLdapSchema {
+  main::runAsZimbra("/opt/zimbra/libexec/zmldapschema 2>/dev/null");
+}
+
+if (!$newinstall) {
```

Comment:
is this check required, as $newinstall is defaulted to 0 at line 8 Or is it getting updated somewhere?

Reply:
done!